suricata/update/rule.py

```diff
@@ -282,6 +285,13 @@ def parse(buf, group=None):
     if rule["msg"] is None:
         rule["msg"] = ""
 
+    try:
+        if not rule["sid"]:
+            raise BadSidError("Sid cannot be of type null")
+    except BadSidError as err:
+        logger.error("Failed to parse rule: %s: %s", buf.rstrip(), err)
+        return
+
     rule["raw"] = m.group("raw").strip()
```

Comment:
I think the above should just be:
```
    if not rule["sid"]:
        raise BadSidError("Sid cannot be of type null")
```

`parse_fileobj` will take care logging the error and continuing on.  I think its important that `parse()` raise this error to the caller as `None` implies the provided input didn't look like a rule.  But if it did look like a rule and its was missing a sid, we should raise an error.

Optional: check that the sid can also be parsed as an integer and raise the exception if it can't.